typed/merge.go

```diff
@@ -290,11 +300,9 @@ func (w *mergingWalker) indexListPathElements(t *schema.List, list value.List) (
 			// this element.
 			continue
 		}
-		if _, found := observed.Get(pe); found {
-			errs = append(errs, errorf("duplicate entries for key %v", pe.String())...)
-			continue
+		if _, found := observed.Get(pe); !found {
+			observed.Insert(pe, child)
```

Comment:
Is this existence check necessary? Is `observed` a set?

Reply:
observed is a map from pathelement to the actual value.

the value that ends-up being in the PE is the one we're going to use to merge I think, so if we were to override here, I think we would end-up merging with the last item in the list rather than the first. But since we've decided we might not want to do that, I will probably deal with that differently.